### PR TITLE
ICU-22054 Remove declarations for unimplemented APIs

### PR DESCRIPTION
**Pull request by cyndyishida**
This patch also includes marking `=delete` on specific `normal` member functions, as opposed to compiler generated functions, based on the description of such functions' surrounding comments.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22054
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable (current build & tests should suffice)
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by richgillam — COMMENTED** [review]
This all looks fine to _me_, but I feel like we should have one or more of the long-timers here, such as @markusicu , weigh in on whether any of these really _should_ have implementations (or some other treatment?), since I don't have the history.

**Comment by cyndyishida** [comment]
> This all looks fine to _me_, but I feel like we should have one or more of the long-timers here, such as @markusicu , weigh in on whether any of these really _should_ have implementations (or some other treatment?), since I don't have the history.

Thanks Rich! I agree a blanket approach to all of these declarations may not be the right solution. I've added Mark, please let me know of others who should take a look as well.

File: icu4c/source/i18n/unicode/vtzone.h
Comment (yumaoka) [inline icu4c/source/i18n/unicode/vtzone.h:385]: Looks good to me.

File: icu4c/source/i18n/unicode/numberrangeformatter.h
Comment (sffc) [inline icu4c/source/i18n/unicode/numberrangeformatter.h:563]: This is a private function. OK to remove.

File: icu4c/source/i18n/unicode/numberrangeformatter.h
Comment (sffc) [inline icu4c/source/i18n/unicode/numberrangeformatter.h:699]: This is a private function. OK to remove.

File: icu4c/source/i18n/unicode/listformatter.h
Comment (sffc) [inline icu4c/source/i18n/unicode/listformatter.h:274]: This is a private function. OK to remove.

File: icu4c/source/i18n/unicode/numsys.h
Comment (sffc) [inline icu4c/source/i18n/unicode/numsys.h:214]: These are private functions. OK to remove.

**Review by sffc — COMMENTED** [review]
I reviewed the components I'm most familiar with, and they look good. I left comments on the functions I looked at.

File: icu4c/source/i18n/unicode/regex.h
Comment (aheninger) [inline icu4c/source/i18n/unicode/regex.h:364]: The three changes in this file look good to me.

File: icu4c/source/i18n/regexcmp.h
Comment (aheninger) [inline icu4c/source/i18n/regexcmp.h:160]: You could also take out the forward declaration of RegexTableEl at line 40 in this file.
The changes to this file otherwise look good - I think that they're all just oversights or sloppy maintenance.

File: icu4c/source/i18n/unicode/calendar.h
Comment (markusicu) [reply icu4c/source/i18n/unicode/calendar.h:1146]: We say that we don't remove deprecated APIs. Removing ones that we never implemented is fine, because no one can ever have used them, but removing ones that do have implementations is problematic.

https://unicode-org.github.io/icu/userguide/icu/design.html#icu-api-compatibility
“Use of deprecated APIs is strongly discouraged, but they are retained for backward compatibility.”

File: icu4c/source/i18n/unicode/tzrule.h
Comment (markusicu) [reply icu4c/source/i18n/unicode/tzrule.h:261]: A note to icu-design about removing unimplemented declarations, at least when marked `@stable`, sgtm.

File: icu4c/source/i18n/unicode/vtzone.h
Comment (markusicu) [reply icu4c/source/i18n/unicode/vtzone.h:385]: private anyway...

**Review by markusicu — COMMENTED** [review]
lgtm tnx
with some comments, and please review the comments from others

File: icu4c/source/i18n/unicode/calendar.h
Comment (cyndyishida) [reply icu4c/source/i18n/unicode/calendar.h:1146]: I think this is what @pedberg-icu had brought up during the internal review. Though these are functions that could be implicitly castable from `EDateFields` to `UCalendarDateFields`. I would not consider these implemented since C++ mangling stores the complete parameter typename. The symbols ultimately don't match up.
e.g. 
```
$ c++filt __ZN3icu8Calendar5clearENS0_11EDateFieldsE
icu::Calendar::clear(icu::Calendar::EDateFields)
```

If we want to keep these declarations to avoid potentially disrupting client code, I agree adding explicit definitions that cast and call the non-deprecated `UCalendarDateFields` versions is a good approach. What do you think?

Comment (cyndyishida) [reply icu4c/source/i18n/unicode/calendar.h:1146]: This was discussed in this weeks ICU-TC meeting and we are removing these function decls.

File: icu4c/source/i18n/unicode/tzrule.h
Comment (cyndyishida) [reply icu4c/source/i18n/unicode/tzrule.h:261]: Yoshito will be dropping a note to icu-design regarding this removal as well as the deprecated functions that take in `EDateFields`.

**Comment by cyndyishida** [comment]
The PR changes are complete and awaiting approval.

**Review by pedberg-icu — APPROVED** [review]

**Pull request merged**
